tests/virt_autotest/uefi_guest_verification.pm

```diff
@@ -0,0 +1,100 @@
+package uefi_guest_verification;
+
+use base 'virt_feature_test_base';
+use strict;
+use warnings;
+use POSIX 'strftime';
+use File::Basename;
+use testapi;
+use IPC::Run;
+use utils;
+use virt_utils;
+use virt_autotest::common;
+use virt_autotest::utils;
+
+sub run_test {
+    my $self = shift;
+
+    $self->check_guest_bootloader($_) foreach (keys %virt_autotest::common::guests);
+    if (is_kvm_host) {
+        record_soft_failure("In order to implement pm features, current kvm virtual machine uses uefi firmware that does not support PXE/HTTP boot and secureboot", "bsc#1182886 UEFI virtual machine boots with trouble");
+        #$self->check_guest_uefi_boot($_) foreach (keys %virt_autotest::common::guests);
+        #$self->check_guest_secure_boot($_) foreach (keys %virt_autotest::common::guests);
+    }
+    else {
+        record_soft_failure("UEFI implementation for xen fullvirt uefi virtual machine is incomplete", "bsc#1184936 Xen fullvirt lacks of complete support for UEFI");
+    }
+    $self->check_guest_pmsuspend_enabled;
+    return $self;
+}
+
+sub check_guest_bootloader {
+    my ($self, $guest_name) = @_;
+
+    my $ssh_command_prefix = "ssh -vvv -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no";
+    script_retry("$ssh_command_prefix root\@$guest_name efibootmgr -v");
```

Comment:
I do not find the check that the booted os/firmware is uefi(yes, I do see more advanced checks on top of uefi).
What about adding such check in this output? I am not quite familiar with uefi outputs, but I guess this can be one place to check it, from my trying results on my non-uefi machine.

On my machine which does not use uefi installation, the output is:
vh013:~ # efibootmgr -v
EFI variables are not supported on this system.
vh013:~ #

Reply:
Checking had already been added, just the assert_script_run(efibootmgr -v). Your case returns non-ok, so fails.